.github/workflows/_cleanup.yml

```diff
@@ -0,0 +1,56 @@
+name: Setup Workload Identity 
+
+on:
+  workflow_call:
+    inputs:
+      cluster-name:
+        required: true
+        description: AKS Cluster Name With Randomly Generated Suffix
+        type: string
+  workflow_dispatch:
+    inputs:
+      cluster-name:
+        required: true
+        description: AKS Cluster Name With Randomly Generated Suffix
+        type: string
+
+
+jobs:
+  setup-workload-identity:
+    name: Setup Workload Identity 
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Log into Azure
+        run: |
+          az login --service-principal \
+            --username ${{ secrets.AZURE_SERVICE_PRINCIPAL_APP_ID }} \
+            --password ${{ secrets.AZURE_SERVICE_PRINCIPAL_PASSWORD }} \
+            --tenant ${{ secrets.AZURE_SERVICE_PRINCIPAL_TENANT }}
+            
+      - name: Install Dependencies 
+        id: install-dependencies 
+        run: |
+
+            curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+            chmod +x kubectl
+            sudo mv kubectl /usr/local/bin
+            kubectl version 2>&1 || true
+
+      - name: Cleanup
+        id: cleanup
+        env: 
+          RESOURCE_GROUP: ${{ vars.RESOURCE_GROUP }} 
+          CLUSTER_NAME: ${{ inputs.cluster-name }}
+        run: | 
+          az aks get-credentials --name $CLUSTER_NAME --resource-group $RESOURCE_GROUP  --overwrite-existing
+          kubectl delete pod -n kafka kafka-producer 2>&1 || true 
```

Comment:
All of these deletes should be covered by just deleting the resource group

Reply:
The resource group and identity in the resource group used for setting up federated id should be static so that we do not have to re-issue role access each time before pipeline runs.

Reply:
I misread line 56. You shouldn't have to delete the pod, etc. if you're going to delete the cluster at the end. So just stop & delete the cluster.